File: BackEnd/XeGo-Microservices/Services/Ride/XeGo.Services.Ride.API/Entities/Ride.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using XeGo.Shared.Lib.Constants;
using XeGo.Shared.Lib.Entities;

namespace XeGo.Services.Ride.API.Entities
{
    public class Ride : BaseEntity
    {
        [Key][DatabaseGenerated(DatabaseGeneratedOption.Identity)] public int Id { get; set; }
        [Required] public string RiderId { get; set; } = string.Empty!;
        public string? DriverId { get; set; }
        public int? CouponId { get; set; }
        [Required] public int VehicleId { get; set; }
        [Required] public string Status { get; set; } = RideStatusConstants.FindingDriver;
        [Required] public double StartLatitude { get; set; }
        [Required] public double StartLongitude { get; set; }
        [Required] public string StartAddress { get; set; } = String.Empty!;
        [Required] public double DestinationLatitude { get; set; }
        [Required] public double DestinationLongitude { get; set; }
        [Required] public string DestinationAddress { get; set; } = String.Empty!;
        [Required] public DateTime PickupTime { get; set; } = DateTime.UtcNow;
        [Required] public bool IsScheduleRide { get; set; } = false;
        public string? CancelledBy { get; set; }
        public string? CancellationReason { get; set; }
    }
}

File: FrontEnd/xego_rider/lib/models/Entities/ride.dart
class Ride {
  final int id;
  final String riderId;
  String? driverId;
  int? discountId;
  int? vehicleId;
  final int vehicleTypeId;
  final String status;
  final double startLatitude;
  final double startLongitude;
  final String startAddress;
  final double destinationLatitude;
  final double destinationLongitude;
  final String destinationAddress;
  final DateTime pickupTime;
  final bool isScheduleRide;
  String? cancelledBy;
  String? cancellationReason;

  Ride({
    required this.id,
    required this.riderId,
    this.driverId,
    this.discountId,
    this.vehicleId,
    required this.vehicleTypeId,
    required this.status,
    required this.startLatitude,
    required this.startLongitude,
    required this.startAddress,
    required this.destinationLatitude,
    required this.destinationLongitude,
    required this.destinationAddress,
    required this.pickupTime,
    required this.isScheduleRide,
    this.cancelledBy,
    this.cancellationReason,
  });

  factory Ride.fromJson(Map<String, dynamic> json) {
    return Ride(
      id: json['id'],
      riderId: json['riderId'],
      driverId: json['driverId'],
      discountId: json['discountId'],
      vehicleId: json['vehicleId'],
      vehicleTypeId: json['vehicleTypeId'],
      status: json['status'],
      startLatitude: json['startLatitude'],
      startLongitude: json['startLongitude'],
      startAddress: json['startAddress'],
      destinationLatitude: json['destinationLatitude'],
      destinationLongitude: json['destinationLongitude'],
      destinationAddress: json['destinationAddress'],
      pickupTime: DateTime.parse(json['pickupTime']),
      isScheduleRide: json['isScheduleRide'],
      cancelledBy: json['cancelledBy'],
      cancellationReason: json['cancellationReason'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'riderId': riderId,
      'driverId': driverId,
      'discountId': discountId,
      'vehicleId': vehicleId,
      'vehicleTypeId': vehicleTypeId,
      'status': status,
      'startLatitude': startLatitude,
      'startLongitude': startLongitude,
      'startAddress': startAddress,
      'destinationLatitude': destinationLatitude,
      'destinationLongitude': destinationLongitude,
      'destinationAddress': destinationAddress,
      'pickupTime': pickupTime.toIso8601String(),
      'isScheduleRide': isScheduleRide,
      'cancelledBy': cancelledBy,
      'cancellationReason': cancellationReason,
    };
  }
}
